monoio/src/buf/io_vec_buf.rs

```diff
@@ -175,11 +175,11 @@ unsafe impl IoVecBufMut for VecBuf {
         for (idx, iovec) in self.iovecs.iter_mut().enumerate() {
             if iovec.iov_len <= len {
                 // set_init all
-                self.raw[idx].set_len(iovec.iov_len);
+                self.raw[idx].set_init(iovec.iov_len);
```

Comment:
Not sure on this behaviour. Not all `IoBufMut` mutate buffer for `set_init`
Notably, only `Vec<u8>` and `bytes::BytesMut` are mutated by `IoBufMut::set_init`

Reply:
We need write back the length from uring&kernel. So all `BufMut` types must be able to set length.
So `unsafe` is necessary!